lib/ansible/modules/storage/hpe3par/hpe3par_cpg.py

```diff
@@ -0,0 +1,341 @@
+
+
+def cpg_ldlayout_map(ldlayout_dict):
+    if ldlayout_dict['RAIDType'] is not None and ldlayout_dict['RAIDType']:
+        ldlayout_dict['RAIDType'] = client.HPE3ParClient.RAID_MAP[
+            ldlayout_dict['RAIDType']]['raid_value']
+    if ldlayout_dict['HA'] is not None and ldlayout_dict['HA']:
+        ldlayout_dict['HA'] = getattr(
+            client.HPE3ParClient, ldlayout_dict['HA'])
+    return ldlayout_dict
+
+
+def create_cpg(
+        client_obj,
+        cpg_name,
+        domain,
+        growth_increment,
+        growth_increment_unit,
+        growth_limit,
+        growth_limit_unit,
+        growth_warning,
+        growth_warning_unit,
+        raid_type,
+        set_size,
+        high_availability,
+        disk_type):
+    try:
+        if not validate_set_size(raid_type, set_size):
+            return (False, False, "Set size not part of RAID set", {})
+        if not client_obj.cpgExists(cpg_name):
+            ld_layout = dict()
+            disk_patterns = []
+            if disk_type is not None and disk_type:
+                disk_type = getattr(client.HPE3ParClient, disk_type)
+                disk_patterns = [{'diskType': disk_type}]
+            ld_layout = {
+                'RAIDType': raid_type,
+                'setSize': set_size,
+                'HA': high_availability,
+                'diskPatterns': disk_patterns}
+            ld_layout = cpg_ldlayout_map(ld_layout)
+            if growth_increment is not None:
+                growth_increment = convert_to_binary_multiple(
+                    growth_increment, growth_increment_unit)
+            if growth_limit is not None:
+                growth_limit = convert_to_binary_multiple(
+                    growth_limit, growth_limit_unit)
+            if growth_warning is not None:
+                growth_warning = convert_to_binary_multiple(
+                    growth_warning, growth_warning_unit)
+            optional = {
+                'domain': domain,
+                'growthIncrementMiB': growth_increment,
+                'growthLimitMiB': growth_limit,
+                'usedLDWarningAlertMiB': growth_warning,
+                'LDLayout': ld_layout}
+            client_obj.createCPG(cpg_name, optional)
+        else:
+            return (True, False, "CPG already present")
+    except exceptions.ClientException as e:
+        return (False, False, "CPG creation failed | %s" % (e))
+    return (True, True, "Created CPG %s successfully." % cpg_name)
+
+
+def delete_cpg(
+        client_obj,
+        cpg_name):
+    try:
+        if client_obj.cpgExists(cpg_name):
+            client_obj.deleteCPG(cpg_name)
+        else:
+            return (True, False, "CPG does not exist")
+    except exceptions.ClientException as e:
+        return (False, False, "CPG delete failed | %s" % e)
+    return (True, True, "Deleted CPG %s successfully." % cpg_name)
+
+
+def main():
+
+    module = AnsibleModule(argument_spec=hpe3par.cpg_argument_spec())
```

Comment:
`check_mode` isn't supported, the documentation should mention that information in `notes` ([see](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#documentation-block))

Reply:
Done

Reply:
Added check_mode in the notes